numba/core/annotations/ipython_magic.py

```diff
@@ -0,0 +1,85 @@
+import hashlib
+import os
+import re
+import importlib.util
+import inspect
+
+from IPython.core import magic_arguments
+from IPython.core.magic import Magics, magics_class, cell_magic
+from IPython.paths import get_ipython_cache_dir
+from numba.core.annotations.pretty_annotate import Annotate
+
+
+@magics_class
+class NumbaMagics(Magics):
+    """
+    Numba magic commands
+    """
+
+    @magic_arguments.magic_arguments()
+    @magic_arguments.argument(
+        '-a', '--annotate', action='store_const', const='default',
+        dest='annotate', help="Produce a colorized HTML version of the source."
+    )
+    @cell_magic
+    def numba(self, line, cell):
+        """Compile the jitted function and display the
+        compiled code, control, flow graph, etc.
+        """
+        args = magic_arguments.parse_argstring(self.numba, line)
+        if args.annotate:
+            function_names = self._get_function_names(cell)
+            code = self._preprocess_cell(cell)
+            spec = self._build_module_spec(line, code)
+
+            # Import the module
+            numba_module = importlib.util.module_from_spec(spec)
+            spec.loader.exec_module(numba_module)
```

Comment:
This seems to execute the module in its own namespace - What happens if it refers to globals in the notebook that aren't defined in the module?